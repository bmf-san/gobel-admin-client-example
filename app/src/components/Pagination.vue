<template>
  <ul>
    <li v-for="i of pagecount" :key="i">
      <router-link
        :to="{ name: name, query: { page: i, limit: limit } }"
        :class="{ current: isCurrent(i) }"
        >{{ i }}</router-link
      >
    </li>
  </ul>
</template>

<script>
export default {
  name: "Pagination",
  props: {
    name: String,
    page: Number,
    limit: Number,
    pagecount: Number,
    refresh: Function
  },
  methods: {
    isCurrent(i) {
      return i == this.page;
    }
  }
};
</script>

<style scoped>
.current {
  color: red;
}
</style>
