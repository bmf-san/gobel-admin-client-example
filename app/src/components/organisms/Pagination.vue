<template>
  <div class="pagination">
    <ul class="pagination-list">
      <li v-for="i of pagecount" :key="i">
        <router-link
          :to="{ name: name, query: { page: i, limit: limit } }"
          :class="{ current: isCurrent(i) }"
          ><span>{{ i }}</span></router-link
        >
      </li>
    </ul>
  </div>
</template>

<script>
import { defineComponent } from "vue";

export default defineComponent({
  name: "Pagination",
  props: {
    name: {
      type: String,
      required: true,
    },
    page: {
      type: Number,
      required: true,
      default: 1,
    },
    limit: {
      type: Number,
      required: true,
      default: 10,
    },
    pagecount: {
      type: Number,
      required: true,
      default: 10,
    },
  },
  setup(props) {
    const isCurrent = (i) => {
      return i == props.page;
    };
    return {
      isCurrent,
    };
  },
});
</script>

<style scocped>
.current {
  background: var(--main-color);
  color: #fff !important;
}
</style>
