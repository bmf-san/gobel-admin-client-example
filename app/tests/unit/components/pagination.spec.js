import { shallowMount } from "@vue/test-utils";
import Pagination from "@/components/Pagination.vue";

describe("Pagination.vue", () => {
  it("init", () => {
    const wrapper = shallowMount(Pagination);
    expect(wrapper.isVueInstance).toBeTruthy();
  });

  it("matched snapshot", () => {
    const wrapper = shallowMount(Pagination);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
