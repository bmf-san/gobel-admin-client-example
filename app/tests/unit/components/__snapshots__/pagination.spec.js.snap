// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination.vue matched snapshot 1`] = `<ul></ul>`;
